### ISSUE TITLE
Fix `main` test asset overwrite sequence order

### ISSUE THREAD

**Issue by timothy-nunn** [ukaea/PROCESS#3233]
Currently, the `main` test assets are overwritten by the `tracking` job. However, the regression tests run in parallel to this line of job execution. Therefore, if a merge into main causes changes in the test assets, the main pipeline fails because the test assets are not yet overwritten. 

The regression tests should wait for the overwrite jobs to run only on main, is this possible?
